### PR TITLE
Task/convert to json

### LINKED ISSUE #4: Frends.Excel.ConvertToJson
This issue should be done using the current task as the baseline (https://github.com/CommunityHiQ/Frends.Community.ConvertExcelFile). Note the ConvertExcelFile is quite complex and here we aim to simplify that codebase and make new additions, thus having one output format per task seems to be more logical.

However, the following should be noted:
- The task should have its own NuGet
- The task should only output the output JSON as a plain string
- The task has to have good unit test coverage
- No integration tests are required besides just testing in unit tests
- Some performance tests would be nice in the docs to indicate how it handles small/big files
- We need to add and check SonarCube linting and go through warnings on that
- Need to setup actions workflows
- Update the readme with new task names, testing section, and such

### PR DESCRIPTION
**Pull request by jefim**
Resolved #4

### CODE REVIEW

File: Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/DateFormats.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿namespace Frends.Excel.ConvertToJSON.Definitions;
```
Comment (Svenskapojkarna) [inline Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/DateFormats.cs:1]: Should the namespace be Frends.Excel.ConvertToJSON?

File: Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Input.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace Frends.Excel.ConvertToJSON.Definitions;
```
Comment (Svenskapojkarna) [inline Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Input.cs:4]: Same here.

File: Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Options.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+﻿using System.ComponentModel;
+using System.ComponentModel.DataAnnotations;
+
+namespace Frends.Excel.ConvertToJSON.Definitions;
```
Comment (Svenskapojkarna) [inline Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Options.cs:4]: Same here.

File: Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Result.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
+﻿using System.ComponentModel;
+
+namespace Frends.Excel.ConvertToJSON.Definitions;
```
Comment (Svenskapojkarna) [inline Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Result.cs:3]: Same here.

File: Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Result.cs [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    /// False if conversion fails.
+    /// </summary>
+    [DefaultValue("false")]
+    public bool Success { get; set; }
+
+    /// <summary>
+    /// Exception message.
+    /// </summary>
+    [DefaultValue("")]
+    public string ErrorMessage { get; private set; }
```
Comment (Svenskapojkarna) [inline Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Result.cs:17]: What is the reason for private set?

File: Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/DateFormats.cs (hunk 1 above)
Comment (jefim) [reply Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/DateFormats.cs:1]: In C# best practices namespaces go according to root namespace + folder structure.

File: Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Input.cs (hunk 2 above)
Comment (jefim) [reply Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Input.cs:4]: In C# best practices namespaces go according to root namespace + folder structure.

File: Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Options.cs (hunk 3 above)
Comment (jefim) [reply Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Options.cs:4]: In C# best practices namespaces go according to root namespace + folder structure.

File: Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Result.cs (hunk 4 above)
Comment (jefim) [reply Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Result.cs:3]: In C# best practices namespaces go according to root namespace + folder structure.

File: Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Result.cs (hunk 5 above)
Comment (jefim) [reply Frends.Excel.ConvertToJSON/Frends.Excel.ConvertToJSON/Definitions/Result.cs:17]: Mostly for anyone not being able to overwrite. This way the value is read-only from outside (and the intention is that no one can set this value even by mistake from outside). Setter is still needed though for class private usage in ctor.

**Pull request merged**